sirepo/sim_data/__init__.py

```diff
@@ -392,7 +392,7 @@ def model_defaults(cls, name):
         return res
 
     @classmethod
-    def parse_jid(cls, data):
+    def parse_jid(cls, data, uid=None):
```

Comment:
I have [added similar code](https://github.com/radiasoft/sirepo/compare/issue/2133#diff-40887cde02088c25476cf3326a42dbd3R742), which works around not having a [proper "substitute user" feature](https://github.com/radiasoft/sirepo/issues/2142)

Leave a TODO saying this is temporary and should not be used. We need to fix auth.logged_in_user.

Reply:
I've added a TODO. 

Unfortunately I've run into the issue again. The [function for determining the computeJobHash has changed](https://github.com/radiasoft/sirepo/compare/45a1c80..445e2f6#diff-624152e6f0c299515665e9821242aab5R65) (comparison of prod vs master) such that we can't use the computeJobHash found inside of in.json. We need to calculate it from the data. But, calling sim_data.compute_job_hash() requires us to be in the flask app context and have an authenticated user.
```
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/pkcli/db.py", line 127, in upgrade_runner_to_job_db
    _create_supervisor_state_file(pkio.py_path(f.dirname))
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/pkcli/db.py", line 81, in _create_supervisor_state_file
    computeJobHash=c.compute_job_hash(i), # TODO(e-carlin): Another user cookie problem
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/sim_data/__init__.py", line 164, in compute_job_hash
    cls.lib_file_basenames(data))
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/sim_data/__init__.py", line 163, in <genexpr>
    (str(cls.lib_file_abspath(b, data=data).mtime()) for b in sorted(
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/sim_data/__init__.py", line 256, in lib_file_abspath
    p = cls._lib_file_abspath(basename, data=data)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/sim_data/__init__.py", line 584, in _lib_file_abspath
    simulation_db.simulation_lib_dir(cls.sim_type()).join(basename)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.py", line 688, in simulation_lib_dir
    return simulation_dir(simulation_type).join(_LIB_DIR)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.py", line 673, in simulation_dir
    d = _user_dir().join(sirepo.template.assert_sim_type(simulation_type))
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/simulation_db.py", line 1121, in _user_dir
    uid = sirepo.auth.logged_in_user()
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 198, in logged_in_user
    res = _get_user()
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 498, in _get_user
    return cookie.unchecked_get_value(_COOKIE_USER)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/cookie.py", line 94, in unchecked_get_value
    return _state().get(key, default)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/cookie.py", line 200, in _state
    return flask.g.sirepo_cookie
  File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/werkzeug/local.py", line 348, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/werkzeug/local.py", line 307, in _get_current_object
    return self.__local()
  File "/home/vagrant/.pyenv/versions/2.7.16/envs/py2/lib/python2.7/site-packages/flask/globals.py", line 45, in _lookup_app_object
    raise RuntimeError(_app_ctx_err_msg)
RuntimeError: Working outside of application context.
```

This one is less easier to hack around. Should we solve the problem?

Reply:
Here's a hack that will work for now:

```py
--- a/sirepo/job_supervisor.py
+++ b/sirepo/job_supervisor.py
@@ -92,7 +92,10 @@ def init():
                 'upgrade_runner_to_job_db',
                 _DB_DIR,
             ),
-            env=PKDict(os.environ).pkupdate(PYENV_VERSION='py2'),
+            env=PKDict(os.environ).pkupdate(
+                PYENV_VERSION='py2',
+                SIREPO_AUTH_LOGGED_IN_USER='unused',
+            ),
         )
```

And:

```py
--- a/sirepo/pkcli/db.py
+++ b/sirepo/pkcli/db.py
@@ -75,6 +75,7 @@ def upgrade_runner_to_job_db(db_dir):
                 return
             raise
         u = simulation_db.uid_from_dir_name(run_dir)
+        sirepo.auth.cfg.logged_in_user = u
         c = sim_data.get_class(i.simulationType)
         d = PKDict(
             computeJid=c.parse_jid(i, u),
```